spooky-action.Rmd

```diff
@@ -24,13 +24,17 @@ Deleting variables and creating files on your desktop are obviously surprising e
   operation which should be done with care.
 
 * __Deleting objects in the global environment with `rm(list = ls())`__. 
-  This might seem like a good way to reset the environment so that you're 
+  This might seem like a good way to reset the environment so that your 
   script can run cleanly. But if someone else `source()`s your script, it will
   delete objects that might be important to them. (Of course, you'd hope that
   all of those objects could easily be recreated from another script, but that
   is not always the case).
 
-Because R doesn't constrain the potential scope of functions and scripts, *you* have to. By avoiding these actions, you will create code that is less suprising to other R users. At first, this might seem like tedious busywork. You might find that spooky action is convenient in the moment, and you might convince yourself that it’s necessary or a good idea. But as you share your code with more people and run more code that has been shared with you[^teachers], you'll find spooky action to get more and more surprising and frustrating. 
+Because R doesn't constrain the potential scope of functions and scripts, *you* have to. By avoiding these actions, you will create code that is less surprising to other R users. Moreover, you'll benefit from the experience of other programming pioneers, summarised in the "Principle of Least Astonishment/Surprise" (POLA[^pola]).
+
+[^pola]: https://web.archive.org/web/20181216083109/http://wiki.c2.com/?PrincipleOfLeastAstonishment
```

Comment:
This reference could be updated to https://en.wikipedia.org/wiki/Principle_of_least_astonishment or a new version of the C2Wiki.